src/modules/study/components/ReadingToolbar.tsx

```diff
@@ -40,7 +40,25 @@ export default function ReadingToolbar({
     chapterId: string;
   }>();
   const router = useRouter();
-  const [textSize, setTextSize] = useState(3);
+
+  // Fetch the textSize from local storage, if we're in a browser context
+  const getInitialTextSize = (): number => {
+    if (typeof window === "undefined" || typeof localStorage === "undefined") {
+      return 3;
+    }
+
+    const stored = parseInt(localStorage.getItem("ReadingToolbar-textSize") ?? "", 10);
+    return Number.isNaN(stored) ? 3 : stored;
+  };
+
+  // Persist the textSize in localStorage on update
+  const [textSize, setTextSize] = useState<number>(getInitialTextSize);
+  useEffect(() => {
+    if (typeof window !== "undefined" && typeof localStorage !== "undefined") {
+      localStorage.setItem("ReadingToolbar-textSize", textSize.toString());
+    }
+  }, [textSize]);
```

Comment:
We are going to have a number of settings we want to save in local storage. Let's add a hook to `src/utils/localstorage.ts` that works like this: `const [settings, setSettings] = useLocalStorageState('reading-settings')`

It should work just like `useState` and handles all of the local storage stuff as a JSON object. That way this component can just use the hook and not worry about local storage